cloud-controller-manager/do/kubernetes_lb_validator.go

```diff
@@ -0,0 +1,181 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package do
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/digitalocean/godo"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// LBService represents the v1.service lb object data
+type LBService struct {
+	APIVersion string
+	Kind       string
+	Metadata   Metadata
+	Spec       map[string]interface{}
+	Selector   map[string]interface{}
+	Type       string
+}
+
+// Metadata represents the metadata field in the lb service object
+type Metadata struct {
+	Annotations interface{} `json:"annotation data,omitempty"`
+	Name        string
+	Namespace   string
+}
+
+// DOKSLBServiceValidator validates service type LB
+type KubernetesLBServiceValidator struct {
+	decoder *admission.Decoder
+	Log     logr.Logger
+	GClient *godo.Client
+	Region  string
+}
+
+// DOKSLBServiceValidator ...
+func (v *KubernetesLBServiceValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	svc := &v1.Service{}
+
+	v.Log.V(6).Info("decoding received request")
+	err := v.decoder.Decode(req, svc)
+	if err != nil {
+		v.Log.Error(err, "failed to decode request")
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("failed to decode request: %v", err))
+	}
+
+	v.Log.V(6).Info("checking received request")
+	if svc.Spec.Type != v1.ServiceTypeLoadBalancer || svc.DeletionTimestamp != nil {
+		return admission.Allowed("ignoring the service which is either not a load balancer or is being deleted")
+	}
+
+	// TODO: these forwarding rules are a placeholder. Further development is required to extract the values from the svc object
+	forwardingRules := []godo.ForwardingRule{
+		{
+			EntryProtocol:  "http",
+			EntryPort:      80,
+			TargetProtocol: "http",
+			TargetPort:     80,
+			CertificateID:  "",
+			TlsPassthrough: false,
+		}}
+
+	// check if region env variable was set, if not, retrieve region from metadata service
+	if v.Region == "" {
+		v.Region, err = dropletRegion(v.GClient.Regions)
+		if err != nil {
+			return admission.Errored(http.StatusBadRequest, fmt.Errorf("failed to determine region: %v", err))
+		}
+	}
+
+	lbRequest := v.buildRequest(svc.Name, v.Region, forwardingRules)
+
+	var resp *godo.Response
+
+	// check if old service object exists
+	if req.OldObject.Raw != nil {
+		// decode raw object
+		err = v.decoder.DecodeRaw(req.OldObject, svc)
+		if err != nil {
+			v.Log.Error(err, "failed to decode existing object")
+			return admission.Errored(http.StatusBadRequest, fmt.Errorf("failed to decode request: %v", err))
+		}
+
+		currentLBID := svc.Annotations["kubernetes.digitalocean.com/load-balancer-id"]
+		// perform create if load-balancer-id annotation is missing
+		if currentLBID == "" {
```

Comment:
Please invert the logic: check if the conditions apply to make an update validation request (annotation found and ingress set), and otherwise do a create validation request. This way, we don't need multiple `validateCreate` requests, which is currently the case.